### PR TITLE
Add output mappings, wrapper function regexp, ability to use json plugin as resource file 

### PR DESCRIPTION
**Pull request by ehoogerbeets**
- added the ability to use ilib-loctool-json as a resource file
- added the ability to specify output file mappings, and output file name templates, similar to the other loctool plugins
- updated dependencies
- can now specify a regular expression to match the function wrapper of strings
- Use the logger provided by the loctool instead of using log4js directly (which doesn't work well)
- minimum version of node is now v10

### CODE REVIEW

File: JavaScriptFile.js [hunk 1]
```diff
@@ -33,10 +42,36 @@ var JavaScriptFile = function(props) {
...
     this.logger = this.API.getLogger("loctool.plugin.JavaScriptFile");
     this.set = this.API.newTranslationSet(this.project ? this.project.sourceLocale : "zxx-XX");
+    this.mapping = this.type.getMapping(this.pathName);
+
+    this.localeSpec = props.locale || (this.mapping && this.API.utils.getLocaleFromPath(this.mapping.template, this.pathName)) || "en-US";
+    this.locale = new Locale(this.localeSpec);
+
+    // get the regexp that finds the function call that wraps strings to translate
+    var jsSettings = this.project.settings && this.project.settings.javascript;
+    var wrapper = (jsSettings && jsSettings.wrapper) || "(^R|\\WR)B\\s*\\.\\s*getString(JS)?";
```
Comment (gouniLee) [inline JavaScriptFile.js:56]: I think it needs an additional test case for that case different regular expression than the default one, and make sure if it works ok. what do you think?
What if the user wants the default one and an additional case? for example, webos-javascript plugins parse both getString and $L

Comment (gouniLee) [inline JavaScriptFile.js:56]: I think it needs an additional test case for that case different regular expression than the default one, and make sure if it works ok. what do you think?
What if the user wants the default one and an additional case? for example, webos-javascript plugins parse both getString and $L

Comment (ehoogerbeets) [reply JavaScriptFile.js:56]: I did that sort of test in the loctool samples, but we can do it here as a unit test as well.

Comment (ehoogerbeets) [reply JavaScriptFile.js:56]: For both of them, you would use alternation:

```json
{
    "wrapper": "(\\.\\s*getString(JS)?|\\$L)"
}
```

Comment (ehoogerbeets) [reply JavaScriptFile.js:56]: Actually, there are already unit tests for it at the end of testJavaScriptFile.js: `testJavaScriptFileParseWithAlternateWrapper` and `testJavaScriptFileParseWithAlternateWrapperAndEmbeddedQuote`

Comment (gouniLee) [reply JavaScriptFile.js:56]: oh I see.

**Review by gouniLee — APPROVED** [review]

**Review by soygitana — APPROVED** [review]

**Pull request merged**